### DISCUSSION TITLE
Automatic math preview based on cursor's position

### DISCUSSION THREAD

**Discussion by sleepymalc** [James-Yu/LaTeX-Workshop#3816] [Q&A]
I'm wondering whether it's possible to automatically pop up the preview window when we edit in the math environment. I saw that someone is actually implementing this in emacs, so I'm wondering whether this is possible. 

Additionally, the preview doesn't recognize self-defined command, and I'm wondering whether it's possible to fix that (by importing self-defined commands, or something like this?)

**Comment by James-Yu** [comment]
For the latter question, it's not quite likely if the current implementation won't work. Yet indeed we have some implementations to include newcommand defs in preview.

**Reply by sleepymalc** [reply]
Do you mean you're working on it or it's already out?

**Reply by James-Yu** [reply]
It was out for years iirc.
